src/AtomShell/window.jl

```diff
@@ -154,71 +163,116 @@ flashframe(win::Window, on = true) =
   @dot_ win flashFrame($on)
 
 """
-    progress(win::Window, p=-1)
+    progress!(win::Window, p=-1)
 
 Sets progress value in progress bar. Valid range is [0, 1.0]. Remove progress
 bar when progress < 0; Change to indeterminate mode when progress > 1.
 
 https://github.com/electron/electron/blob/master/docs/api/browser-window.md#winsetprogressbarprogress-options
 """
-progress(win::Window, p = -1) =
-  @dot_ win setProgressBar($p)
+progress!(win::Window, p) = @dot_ win setProgressBar($p)
+
+# Deprecated
+progress(win::Window, progress=-1) = progress!(win, progress)
 
 """
-    title(win::Window, title)
+    title!(window, title)
 
-Set `win`'s title to `title.`
+Set the window's title.
 """
-title(win::Window, title) =
-  @dot_ win setTitle($title)
+title!(win::Window, title) = @dot_ win setTitle($title)
+
+# Deprecated
+title(win::Window, title) = title!(win, title)
 
 """
-    title(win::Window)
+    title(window)
 
 Get the window's title.
 """
-title(win::Window) =
-  @dot win getTitle()
+title(win::Window) = @dot win getTitle()
+
+"""
+    center!(win::Window)
+
+Center a window.
+"""
+center!(win::Window) = @dot_ win center()
 
+# Deprecated (and misspelled?)
 centre(win::Window) =
   @dot_ win center()
 
-position(win::Window, x, y) =
-  @dot_ win setPosition($x, $y)
+"""
+    position!(window, x, y)
+    position!(window, position)
+
+Position a window.
+
+This positions the top-left corner of the window to match the given coordinates.
+"""
+position!(w::Window, x, y) = @dot_ w setPosition($x, $y)
+position!(w::Window, pos) = position!(w, pos...)
+
+# Deprecated
+position(w::Window, x, y) = position!(w, x, y)
+
+"""
+    position(window)
+
+Get the window's position.
+
+This returns a tuple that represents the position of the top-left corner of the
+window.
+"""
+function position(win::Window)
+    x, y = Int.(@dot win getPosition())
+    return (x, y)
+end
 
-position(win::Window) =
-  @dot win getPosition()
+"""
+    resize!(window::$(repr(Window)), width, height)
+    resize!(window::$(repr(Window)), dims)
+
+Resize a window to the given dimensions.
+"""
+Base.resize!(window::Window, width, height) = @dot_ window setSize($width, $height)
+Base.resize!(window::Window, dims) = resize!(window, dims...)
 
-size(win::Window, w::Integer, h::Integer) =
-  invoke(size, Tuple{Window, Any, Any}, win, w, h)
+# Deprecated
+Base.size(window::Window, width, height) = resize!(window, width, height)
 
-size(win::Window, w, h) =
-  @dot_ win setSize($w, $h)
+"""
+    size(window::$(repr(Window)))
```

Comment:
haha same here. I guess this was an attempt to generate `Blink.Window` or `MyPackage.Blink.Window` or w/e? Seems cool, i haven't seen anyone do that before

Reply:
I think it was but in hindsight I think it's just gonna be best to stick to `::Window` and maybe use DocStringExtensions at some point. I'm just gonna use `window::Window` to be consistent. :^)